src/index.c

```diff
@@ -659,18 +575,14 @@ void trimUnionIterator(IndexIterator *iter, size_t offset, size_t limit, bool as
     UI_SyncIterList(ui);
     // todo:
   }
-  if (unsort) {
-    iter->Read = UI_ReadUnsorted;
-  }
+  iter->Read = UI_ReadUnsorted;
```

Comment:
Shouldn't this PR drops all the ReadUnsorted kind of readers?

Reply:
The optimizer still utilize the union iterator unsorted mode. That the only one that I kept. It also didn’t rely on the criteria. Just reading from each child at a time (may be the behavior we always want with numeric ranges

Reply:
But if the other iterators can not handle unsorted results, how can we use an unsorted iterator?

Reply:
Right. In the usual case we can’t and we don’t. But this function is still used (by the optimizer only)